engine/src/main/java/org/pentaho/di/www/StatusServletUtils.java

```diff
@@ -30,6 +30,10 @@
 
 public class StatusServletUtils {
 
+  public static final String RESOURCES_PATH = "/content/common-ui/resources/themes";
+  public static final String STATIC_PATH = "/static";
+  public static final String PENTAHO_ROOT = "/pentaho";
```

Comment:
This would still cause problems ( i.e. broken links ) on the following 2 situations:
- _"old"_ DI-Servers ( a.k.a ) flavourized Pentaho Servers, that look-&-feel as the "old" DI-Servers
- OEM specific webapp naming; if the webapp gets rebranded from `pentaho` to `my-oem-webapp`, `kettle/status` page icons could potentially cease to work

Pentaho platform, for example, provides the `fully-qualified-server-url` property via `server.properties` ( which is a user configurable file ), for customers to tweak / change the base server endpoint to be used for things such as static resources.

https://github.com/pentaho/pentaho-platform/blob/8.2.0.0-RC/assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/server.properties#L3

Also providing a one-liner to fetch it: `PentahoSystem.getApplicationContext().getFullyQualifiedServerURL();`
- Example: https://github.com/pentaho/pentaho-platform/blob/8.2.0.0-RC/extensions/src/main/java/org/pentaho/platform/web/servlet/PluginDispatchServlet.java#L356

Could it be that, when `!isJettyMode()` , `prefix` should be `PentahoSystem.getApplicationContext().getFullyQualifiedServerURL()` ?

Reaching out to @mbatchelor @pamval on this for confirmation / double-checking.

Reply:
@pedrofvteixeira -- to speak to your second concern, your suggestion is probably a better implementation. However, I do address this concern by getting the "base server endpoint" as follows:

String root = request.getRequestURI() == null ? StatusServletUtils.PENTAHO_ROOT : request.getRequestURI().substring( 0, request.getRequestURI().indexOf( CONTEXT_PATH ) );
String prefix = isJettyMode() ? StatusServletUtils.STATIC_PATH : root + StatusServletUtils.RESOURCES_PATH;

Reply:
@pedrofvteixeira @ddiroma are all concerns addressed? Can someone hit the "resolve conversation" button.

Reply:
Good by me.